odoo/addons/base/models/ir_model.py

```diff
@@ -1069,6 +1070,7 @@ class IrModelSelection(models.Model):
     value = fields.Char(required=True)
     name = fields.Char(translate=True, required=True)
     sequence = fields.Integer(default=1000)
+    ondelete = fields.Char()
```

Comment:
Just wondering, what is the point of reflecting `ondelete`?
I don't see any added value.

Reply:
@rco-odoo In my original implementation the `_process_ondelete` method would lookup the reflected value and not the value in the registry, but I think you changed it to use the value from the registry, so now it has no use, I will remove the reflection logic

Reply:
Oh wait, now I remember.  The changes I did are valid for processing a module uninstallation.
But they are not valid for processing a module upgrade where one value in the selection has been removed.  In that case, one has to keep the old value in the `ondelete`.  Does that sound right?

For instance, assume we have in version 13:
```python
state = fields.Selection([('foo', 'Foo'), ('bar', 'Bar'), ('baz', 'Baz')],
                         required=True, default='foo')
```
Then in version 14, one would need:
```python
state = fields.Selection([('foo', 'Foo'), ('bar', 'Bar')],
                         ondelete={'baz': 'set default'},
                         required=True, default='foo')
```
The latter will typically be replaced by a migration script.  But then, the deletion of the `ir.model.fields.selection` must not crash.

Reply:
The latter should be handled in a migration script as you said, first by setting the existing records with the option to be deleted to a valid option / unlinking said records then removing the selection option from `ir.model.fields.selection`.

I don't think this will crash, as in this case the `_get_records` will get an empty recordset if the migration script was done in pre, and then we'll try to write / unlink / whatever ib that empty recorset which is a no-op (right?)